indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -244,7 +297,9 @@ public TaskStatus call()
                                 command.add(nodeType);
                               }
 
-                              jsonMapper.writeValue(taskFile, task);
+                              if (!taskFile.exists()) {
```

Comment:
do you need to do any validation on the taskFile?

Example: if its empty or corrupted or simply not equal to the task that we're expecting.